jsonschema-validation.xml

```diff
@@ -1005,7 +1014,14 @@
                 <t>
                     If the instance is a string, this property indicates the media type
                     of the contents of the string.  If "contentEncoding" is present,
-                    this property describes the decoded string.
+                    this property describes the decoded string.  If the "type" keyword is
+                    absent, this keyword MAY be interpreted as describing an unencoded binary
+                    resource.  The exact meaning and behavior of this untyped usage is
+                    application-defined.
```

Comment:
This is effectively permitting values outside the data model, which sort of defeats the point of having a data model, doesn't it?

That's not to say this should be illegal, but maybe we can describe this inside the data model section instead? Something like "JSON Schema can be applied to data that is a superset of this data model; in this case, the instance may be outside any of the six primitive types."

Reply:
@awwright I like your idea of simply extending the data model.  I'd probably recommend that we just add `binary` rather than throwing it entirely open, but I could be persuaded on that point.

What's going on here is that I'm recognizing a thing that a significant number of people (OpenAPI users) were already doing in the wild, in a way that was _more_ problematic.  OAS 3.0 treats binary data as `"type": "string"` plus a custom `format`, which is definitely wrong.  Unencoded binary data cannot be considered a JSON string in any reasonable universe.

Since OAS 3.1 is picking up the `content*` keywords, they were willing to drop the custom `format` and use of `"type": "string"` as long as there was something that they could recommend for this use case.  This is a scenario where nothing's going to be validated, but a JSON Schema is used for descriptive purposes.  Which is fine with `content*` because they don't do validation anyway.  They are annotations, so like all annotations, their behavior is application-defined.  This is particularly unusual behavior so it is called out.  Also, we want it called out so people don't come up with worse "solutions" again.

Reply:
I would argue, let's just call binary for now, and if other use cases present, then open it up later.
I'd like to avoid speculation and stay grounded on use cases right infront of our faces first.
=]

Reply:
@Relequestual @awwright just to clarify, we're now proposing to add `binary` as a type?

I feel like this should get an issue for visibility so that others can comment on it.  It's a major change from what I worked out with OAS.  I'm reasonably OK with making the change, but it's kind of a big deal.

There are also alternatives such as a `"binary": true` approach.  This would avoid breaking compatibility with anything that hardcodes the current set of valid `type` values.

I will file an issue for this.